libs-scala/akka-stream-util/src/main/scala/com/daml/akka/stream/InstrumentedSource.scala

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.akka.stream
+
+import akka.Done
+import akka.stream.scaladsl.{Source, SourceQueueWithComplete}
+import akka.stream.{Materializer, OverflowStrategy, QueueOfferResult}
+import com.codahale.metrics.Counter
+import com.daml.dec.DirectExecutionContext
+
+import scala.concurrent.Future
+
+object InstrumentedSource {
+
+  final class QueueWithComplete[T](
+      delegate: SourceQueueWithComplete[T],
+      saturation: Counter,
```

Comment:
This isn't saturation in the sense I understand it (% out of total capacity), but rather the number of elements queued.

Reply:
I've renamed the counters to be "capacity" (was size) and "length" (was saturation).